File: source/generatereport.php
<?php
$db_hostname = getenv('RDS_HOSTNAME');
$db_username = getenv('RDS_USERNAME');
$db_password = getenv('RDS_PASSWORD');
$db_port = getenv('RDS_PORT');
$prodDbName = 'QISAMStatus';

$serverName = $db_hostname.','.$db_port;

$connectioninfo = array("Database"=>"$prodDbName","UID"=>"$db_username","PWD"=>"$db_password");   
$conn = sqlsrv_connect( $serverName, $connectioninfo);
if( !$conn ) {
	echo "A database connection could not be established. Please contact your administrator<br />";
	die( print_r( sqlsrv_errors(), true));
} 

//$SAM = '1AAA-01';

$Activity5 = sqlsrv_query( $conn, "SELECT [Result1],[Description1],[Result2],[Description2],[Result3],[Description3],[Result4],[Description4],[Result5],[Description5],[Result6],[Description6] FROM [Activity5] WHERE [SAM]='$SAM'");
$Activity8 = sqlsrv_query( $conn, "SELECT [Result1],[Description1],[Result2],[Description2] FROM [Activity8] WHERE [SAM]='$SAM'");
$Activity16 = sqlsrv_query( $conn, "SELECT [Result1],[Description1] FROM [Activity16] WHERE [SAM]='$SAM'");
$Activity17 = sqlsrv_query( $conn, "SELECT [Result1],[Description1] FROM [Activity17] WHERE [SAM]='$SAM'");
$row5 = sqlsrv_fetch_array( $Activity5, SQLSRV_FETCH_ASSOC);
$row8 = sqlsrv_fetch_array( $Activity8, SQLSRV_FETCH_ASSOC);
$row16 = sqlsrv_fetch_array( $Activity16, SQLSRV_FETCH_ASSOC);
$row17 = sqlsrv_fetch_array( $Activity17, SQLSRV_FETCH_ASSOC);

require_once 'Classes/PHPExcel/IOFactory.php';
require_once 'Classes/PHPExcel.php';

$excel2 = PHPExcel_IOFactory::createReader('Excel2007');
$excel2 = $excel2->load('QA_Checklist.xlsx'); 
$excel2->setActiveSheetIndex(0);
$excel2->getActiveSheet()->setCellValue('C28', $row5['Result1'] )
    ->setCellValue('D28', $row5['Description1'])
    ->setCellValue('C29', $row5['Result2'])       
    ->setCellValue('D29', $row5['Description2'])
	->setCellValue('C30', $row5['Result3'] )
	->setCellValue('D30', $row5['Description3'])
	->setCellValue('C31', $row5['Result4'])
	->setCellValue('D31', $row5['Description4'])
	->setCellValue('C32', $row5['Result5'])
	->setCellValue('D32', $row5['Description5'])
	->setCellValue('C33', $row5['Result6'])
	->setCellValue('D33', $row5['Description6'])
	->setCellValue('C62', $row8['Result1'])
	->setCellValue('D62', $row8['Description1'])
	->setCellValue('C63', $row8['Result2'])
	->setCellValue('D63', $row8['Description2'])
	->setCellValue('C99', $row16['Result1'])
	->setCellValue('D99', $row16['Description1'])
	->setCellValue('C101', $row17['Result1'])
	->setCellValue('D101', $row17['Description1']);
	
$objWriter = PHPExcel_IOFactory::createWriter($excel2, 'Excel2007');
$FileName = 'QA_Checklist_'.$SAM.'.xlsx';
$objWriter->save($FileName);

?>

File: source/php/Generatereport.py
import openpyxl                      
import os ,sys


class Generatereport:

    def mainfunction(self,sam):
        book = openpyxl.load_workbook('QA_CheckList.xlsx')
        sheet = book.get_sheet_by_name('Sheet2')
        if os.path.exists('activity17.txt'):
            file = open('activity17.txt','r')
            array =  file.readlines()
            sheet['C101'] = array[0]
            sheet['D101'] = array[1]
            activity17status = array[2]
        else:
            sheet['D101'] = "File Improper"
            
        if os.path.exists('activity15.txt'):
            file = open('activity15.txt', 'r')
            array =  file.readlines()
            sheet['C95'] = array[0]
            sheet['C96']  = array[2]
            sheet['C97']  = array[4]
            sheet['D95'] = array[1]
            sheet['D96']  = array[3]
            sheet['D97'] = array[5]
            activity15status = array[6]
        else:
            sheet['D95'] = "File Improper"
            sheet['D96'] = "File Improper"
            sheet['D97'] = "File Improper"



        if os.path.exists('activity16.txt'):
            file = open('activity16.txt', 'r')
            array =  file.readlines()
            sheet['C99'] = array[0]
            sheet['D99'] = array[1]
            activity16status = array[2]
        else:
            sheet['D99'] = "File Improper"

        if os.path.exists('activity11.txt'):
            file = open('activity11.txt', 'r')
            array =  file.readlines()
            sheet['C74'] = array[0]
            sheet['C76'] = array[2]
            sheet['C78'] = array[4]
            sheet['D74'] = array[1]
            sheet['D76'] = array[3]
            sheet['D78'] = array[5]
            activity11status = array[6]
        else:
            sheet['D74'] = "File Improper"
            sheet['D76'] = "File Improper"
            sheet['D78'] = "File Improper"

        if os.path.exists('activity8.txt'):
            file = open('activity8.txt', 'r')
            array =  file.readlines()
            sheet['C62'] = str(array[0])
            sheet['C63'] = str(array[2])
            sheet['D62'] = str(array[1])
            sheet['D63'] = str(array[3] )
            activity8status = array[4]
        else:                                               
            sheet['D62'] = "File Improper"
            sheet['D63'] = "File Improper"

        if os.path.exists('activity5.txt'):
            file = open('activity5.txt', 'r')
            array =  file.readlines()
            sheet['C29'] = array[0]
            sheet['C30'] = array[2]
            sheet['C31'] = array[4]
            sheet['D29'] = array[1]
            sheet['D30'] = array[3]
            sheet['D31'] = array[5]
            activity5status = array[6]
        else:                                        
            sheet['D29'] = "File Improper"
            sheet['D30'] = "File Improper"
            sheet['D31'] = "File Improper"


        dest_filename = 'C:\\QI_REPORTS\\' + 'QA_Report_'+sam+'.xlsx'
        book.save(dest_filename)

if __name__ == '__main__':
    sam = sys.argv[1]
    try:
        ge = Generatereport()
        ge.mainfunction(sam)
        if os.path.exists('activity17.txt'):
            os.remove('activity17.txt')
        if os.path.exists('activity16.txt'):
            os.remove('activity16.txt')
        if os.path.exists('activity15.txt'):
            os.remove('activity15.txt')
        if os.path.exists('activity11.txt'):
            os.remove('activity11.txt')
        if os.path.exists('activity8.txt'):
            os.remove('activity8.txt')
        if os.path.exists('activity5.txt'):
            os.remove('activity5.txt')
        print 'Success'
    except:
        print 'Failed'
